app/controllers/comments_controller.rb

```diff
@@ -12,14 +12,30 @@ def create
     end
   end
 
+  def edit
+    @comment = Comment.find(params[:id])
+    @review = Review.find(params[:review_id])
+  end
+
+  def update
+    @comment = Comment.find(params[:id])
+    @review = Review.find(params[:review_id])
+    if @comment.update_attributes(comment_params)
+      flash[:notice] = "Comment updated!"
+      redirect_to dinosaur_path(@comment.review.dinosaur_id)
+    else
+      flash[:errors] = @comment.errors.full_messages.join(". ")
+      render :edit
+    end
+  end
+
   protected
 
   def comment_params
     params.require(:comment).permit(
-      :body,
-      :review_id).merge(
+      :body).merge(
         user_id: current_user.id,
-        review_id: review
```

Comment:
I'll take the blame for this one, because I didn't fix it when helping you yesterday, but just pass in:

```
user: current_user,
review: review
```

It's preferable to use whole objects rather than just checking against the existence of the foreign key on a given object.